docs/Design/SoftDetailedDes/MIS.tex

```diff
@@ -229,6 +228,92 @@ \subsubsection{Local Functions}
 
 \newpage
 
+\section{MIS of Location Management Module} \label{mLocation} 
+The Location Management Module manages user location data and outputs relevant resources based the available user location data.
+
+\wss{You can reference SRS labels, such as R\ref{R_Inputs}.}
+
+\wss{It is also possible to use \LaTeX for hypperlinks to external documents.}
+
+\subsection{Module}
+
+locationManagement
+
+\subsection{Uses}
+
+ Database Driver Module (Section \ref{mDBDriver})
+
+\subsection{Syntax}
+
+\subsubsection{Exported Constants}
+None
+
+\subsubsection{Exported Access Programs}
+
+\begin{center}
+\begin{tabular}{p{5cm} p{3cm} p{3cm} p{5cm}}
+\hline
+\textbf{Name} & \textbf{In} & \textbf{Out} & \textbf{Exceptions} \\
+\hline
+getLastLocation & - & String & noLocationSavedException \\
+setLastLocation & String & - & invalidLocationException \\
+getNearbyStores & - & List$<$String$>$ & - \\
+setRadius & $\mathbb{N}$ & - & - \\
+\hline
+\end{tabular}
+\end{center}
+
+\subsection{Semantics}
+
+\subsubsection{State Variables}
+None
+
+\subsubsection{Environment Variables}
+None
+
+\subsubsection{Assumptions}
+None
+
+\subsubsection{Access Routine Semantics}
+
+\noindent getLastLocation():
+\begin{itemize}
+\item transition: none
+\item output: \( \textit{out} \coloneqq location \), where textit{location} is a String representing the user location stored in the database (Section \ref{mDBDriver})
+\item exception: \( \textit{exc} \coloneqq noLocationSavedException \), if the database (Section \ref{mDBDriver}) has no value saved for user location
+\end{itemize}
+
+\noindent setLastLocation(\textit{location}):
+\begin{itemize}
+\item transition: Set user location in the database (Section \ref{mDBDriver}) to \textit{location}
+\item output: none
+\item exception: \( \textit{exc} \coloneqq invalidLocationException \), if \textit{location} is not a valid location in the places dependency
+\end{itemize}
+
+\noindent getNearbyStores():
+\begin{itemize}
+\item transition: none
+\item output: \( \textit{out} \coloneqq locations \), where textit{locations} is a List of Strings representing relevant stores specified by the user's preferences stored in the database (Section \ref{mDBDriver})
```

Comment:
You might be able to describe this using "for all stores in radius", but not 100% sure.

Also, could this method potentially have an exception case if a radius is not set? Or would it just return an empty list of locations (my functions generally follow this latter option where I explicitly write that null is returned if a certain condition is true like `if x text is not found, return null` but in formal language)? 

i.e. `radius preferences = null \Rightarrow noRadiusSetException`? Just a thought I had. Although, you might want to consider a local function for fetching the radius in this case so that you can use a function call instead of `radius preferences` or anything like that which might look better.

Reply:
even if it's an empty list i will return a list, not a null